pandas/core/resample.py

```diff
@@ -1519,9 +1519,24 @@ def _get_period_bins(self, ax):
 
         start = ax.min().asfreq(self.freq, how=self.convention)
         end = ax.max().asfreq(self.freq, how='end')
+        if self.base:
+            start = start.to_timestamp()
+            end = end.to_timestamp()
+
+            p_start, p_end = _get_range_edges(start,
+                                              end,
+                                              self.freq,
+                                              closed=self.closed,
+                                              base=self.base)
+            i = None if self.freq.onOffset(start) else 1
+            j = -1 if self.freq.onOffset(end) else None
+        else:
+            p_start, p_end = start, end
+            i = j = None
```

Comment:
if base is non 0: 
I use the same method to get the base adjusted bin edge labels as for a datetimeindex.
Depending on if the start and end dates are on the relevant freq the edges will be extended away from the true start/end labels. 
The i and j indexers are used on the resultant PeriodIndex to adjust them to the proper start/end. 

If base is 0:
then everything is treated the same as before, the start and end date don't change and no slicing of the resultant PeriodIndex is needed so i and j are set to None since `idx[slice(None, None)]` gives the full index

Everything up to here only accounts for what the labels of resultant resample will be, not the actual content being aggregated. For that the bins edges must be adjusted to reflect the base parameter which is done with `rng -= ((freq_mult - self.base) % freq_mult)`

Reply:
Great!  So now take this explanation and add it as a comment in your code.